cmd/podman/common/specgen.go

```diff
@@ -656,25 +655,52 @@ func FillOutSpecGen(s *specgen.SpecGenerator, c *ContainerCLIOpts, args []string
 }
 
 func makeHealthCheckFromCli(inCmd, interval string, retries uint, timeout, startPeriod string) (*manifest.Schema2HealthConfig, error) {
+	cmdArr := []string{}
+	isArr := false
+	cmdSplit := strings.SplitN(inCmd, " ", 2) // default for compat
```

Comment:
The logic here is reverse of what it should be - do a JSON unmarshal first, and if that returns an error attempt to parse as a string.